Destructuring/key-parameters.lisp

```diff
@@ -67,8 +67,48 @@
                    all-ignorables (append ignorables all-ignorables)))
         finally (return (values all-binds all-ignorables))))
 
+(defmethod key-validation-bindings
+    (client (parameter-group key-parameter-group) argument-variable)
+  (declare (ignore client))
+  (let ((glength-check (gensym "LENGTH-CHECK-DUMMY"))
+        (length-check-form
+          `(unless (evenp (cl:length ,argument-variable))
+             (error "odd number of keyword parameters"))))
+    (if (allow-other-keys parameter-group)
+        (values `((,glength-check ,length-check-form)) `(,glength-check))
+        (let* ((unknowns (gensym "UNKNOWN-KEYWORDS"))
+               (known-keywords
+                 (loop for parameter in (parameters parameter-group)
+                       collect (raw (keyword parameter))))
+               (unknowns-form
+                 `(loop with seen-allow-other-keys-p = nil
+                        with allow-other-keys = nil
+                        for (key value) on ,argument-variable by #'cl:cddr
+                        ;; :allow-other-keys is always acceptable, so we have
+                        ;; to be careful here to never include it in the
+                        ;; unknowns list.
+                        if (eq key :allow-other-keys)
+                          do (unless seen-allow-other-keys-p
+                               (setf allow-other-keys value
```

Comment:
The supplied value is stored here, but I don't see where it is used.

Reply:
OK, I see now that a later commits adds this. Could you squash the fixes into the respective commit?

Reply:
By squash do you mean merging commits that are already committed? I don't actually know how to do that

Reply:
Yes, "squashing" means editing the existing commits such that multiple commits are combined into one.

Not to derail this into a git philosophy discussion, but for reviewing and, later, examination of the the commit history, things are easier if commits are self-contained changes. In this review, for example, I remarked on a problem that was fixed in a later commit I had not seen at that point.

For the changes at hand, I would probably make three commits: "Check macro keyword arguments", "Define lambda list unparser" and "Improve error reporting". [This video](https://techfak.de/~jmoringe/magit-rebase.ogv) illustrates the rebasing process using magit for Emacs. The equivalent command-line invocation is `git rebase --interactive START-COMMIT`.

I can totally understand if you don't want to work this way. In that case, please address (possibly by adding comments instead of code changes) the remaining points raised in the review.

Let me know if I can help with anything.